src/jinaga.ts

```diff
@@ -216,33 +216,78 @@ export class Jinaga {
         return this.factManager.startObserver<U>(references, innerSpecification, resultAdded, true);
     }
 
-    /**
-     * Compute the SHA-256 hash of a fact.
-     * This is a deterministic hash that can be used to identify the fact.
-     * @param fact The fact to hash
-     * @returns The SHA-256 hash of the fact as a base-64 string
-     */
-    static hash<T extends Fact>(fact: T) {
-        const hash = lookupHash(fact);
-        if (hash) {
-            return hash;
-        }
-        const error = this.getFactError(fact);
-        if (error) {
-            throw new Error(`Cannot hash the object. It is not a fact. ${error}: ${JSON.stringify(fact)}`);
-        }
-        const reference = dehydrateReference(fact);
-        return reference.hash;
+    /**
+     * Compute the SHA-256 hash of a fact.
+     * This is a deterministic hash that can be used to identify the fact.
+     * @param fact The fact to hash
+     * @returns The SHA-256 hash of the fact as a base-64 string
+     */
+    static hash<T extends Fact>(fact: T) {
+        const hash = lookupHash(fact);
+        if (hash) {
+            return hash;
+        }
+        const error = this.getFactError(fact);
+        if (error) {
+            throw new Error(`Cannot hash the object. It is not a fact. ${error}: ${JSON.stringify(fact)}`);
+        }
+        const reference = dehydrateReference(fact);
+        return reference.hash;
+    }
+
+    /**
+     * Create a strongly-typed fact reference from a type constructor and hash.
+     * This allows you to create a minimal fact object that can be used with
+     * query, watch, and subscribe APIs when you only have the hash.
+     * 
+     * @param ctor The constructor function with a static Type property
+     * @param hash The SHA-256 hash of the fact as a base-64 string
+     * @returns A fact reference object typed as T
+     */
+    static factReference<T extends Fact>(ctor: new (...args: any[]) => T, hash: string): T {
```

Comment:
Look for an existing generic type helper that encapsulates the constructor pattern. I believe it is used in the model builder among other places.